pkg/steps/multi_stage/gen.go

```diff
@@ -247,6 +248,9 @@ func addSecretWrapper(pod *coreapi.Pod, vpnConf *vpnConf) {
 	container := &pod.Spec.Containers[0]
 	args := container.Args
 	container.Args = make([]string, 0)
+	if isObserver {
+		container.Args = append(container.Args, "--ro-kubeconfig")
```

Comment:
This is wrong here. It can be removed.

Reply:
`entrypoint-wrapper` binary exists in both `.spec.initContainers[1]` and `spec.containers[0]` stanzas, but as an initContainer it is just used as a copy command. It does the real job when the test run, acting as a proxy and doing some hacky stuff like:
- manage kubeconfig
- map `$SHARED_DIR` to its secret
- forward `SIGTERM` and `SIGINT`
- ...